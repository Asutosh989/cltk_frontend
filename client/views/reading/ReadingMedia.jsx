import FlatButton from 'material-ui/FlatButton';

ReadingMedia = React.createClass({

	propTypes: {
	},

	render() {
		return (
			<div className="embedded-media">
				<div className="media-outer">
					<img alt="thumbnail" src="/images/aeneid-demo-image-2.png" />
					<FlatButton
						className="show-in-viewer-button"
						label="Image Viewer"
						icon={<i className="mdi mdi-magnify" />}
					/>
				</div>
				<div className="media-description">
					<span className="media-caption">
						"Folio 45V". <em>The Vergilius Vaticanus
						</em>. Vatican, Biblioteca Apostolica, Cod. Vat. lat. 3225). Rome, Italy. Ca 400 AD.
					</span>

				</div>

			</div>
		);
	},
});
