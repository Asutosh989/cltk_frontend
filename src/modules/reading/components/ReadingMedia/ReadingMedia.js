import React from 'react';
import PropTypes from 'prop-types';
import { Meteor } from 'meteor/meteor';
import { createContainer } from 'meteor/react-meteor-data';
import FlatButton from 'material-ui/FlatButton';

class ReadingMedia extends React.Component {

	getDefaultProps() {
		return {
			mediaId: '',
		};
	}


	render() {
		const mediaItem = this.props.mediaItem;

		let image = {
			caption: '"Folio 45V". <em>The Vergilius Vaticanus</em>. Vatican, Biblioteca Apostolica, Cod. Vat. lat. 3225). Rome, Italy. Ca 400 AD.',
		};
		let imageUrl = '/images/aeneid-demo-image-2.png';

		/*
		if (mediaItem) {
			imageUrl = mediaItem.url();
		}
		*/

		return (
			<div className="embedded-media">
				<div className="media-outer">
					<img alt="thumbnail" src={imageUrl} />
					<FlatButton
						className="show-in-viewer-button"
						label="Image Viewer"
						icon={<i className="mdi mdi-magnify" />}
					/>
				</div>
				<div className="media-description">
					<span className="media-caption">
						{image.caption}
					</span>

				</div>

			</div>
		);
	}
}

ReadingMedia.propTypes = {
	mediaId: PropTypes.string,
};

const ReadingMediaContainer = createContainer(() => {
	const mediaSubscription = Meteor.subscribe('attachments', this.props.mediaId);
	let attachment = null;
	if (mediaSubscription.ready()) {
		attachment = Attachments.findOne({ _id: this.props.mediaId });
	}

	return {
		mediaItem: attachment,
	};
}, ReadingMedia);


export default ReadingMediaContainer;
