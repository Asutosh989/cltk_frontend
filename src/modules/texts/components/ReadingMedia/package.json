{
  "name": "ReadingMedia",
  "version": "0.0.1",
  "private": true,
  "main": "./ReadingMedia.js"
}
