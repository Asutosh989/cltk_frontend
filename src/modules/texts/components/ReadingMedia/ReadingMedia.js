import React from 'react';
import PropTypes from 'prop-types';
import FlatButton from 'material-ui/FlatButton';

import './ReadingMedia.css';

class ReadingMedia extends React.Component {

	render() {
		let image = {
			caption: '"Folio 45V". <em>The Vergilius Vaticanus</em>. Vatican, Biblioteca Apostolica, Cod. Vat. lat. 3225). Rome, Italy. Ca 400 AD.',
		};
		let imageUrl = '/images/aeneid-demo-image-2.png';

		return (
			<div className="embedded-media">
				<div className="media-outer">
					<img alt="thumbnail" src={imageUrl} />
					<FlatButton
						className="show-in-viewer-button"
						label="Image Viewer"
						icon={<i className="mdi mdi-magnify" />}
					/>
				</div>
				<div className="media-description">
					<span className="media-caption">
						{image.caption}
					</span>

				</div>

			</div>
		);
	}
}

ReadingMedia.propTypes = {
	mediaId: PropTypes.string,
};

export default ReadingMedia;
