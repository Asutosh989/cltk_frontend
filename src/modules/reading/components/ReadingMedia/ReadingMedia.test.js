
import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

// component:
import ReadingMedia from './ReadingMedia';

describe('ReadingMedia', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(
				<MuiThemeProvider>
					<ReadingMedia />
				</MuiThemeProvider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
