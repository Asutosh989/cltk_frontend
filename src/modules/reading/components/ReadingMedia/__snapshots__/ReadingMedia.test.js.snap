// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadingMedia renders correctly 1`] = `
<div
  className="embedded-media"
>
  <div
    className="media-outer"
  >
    <img
      alt="thumbnail"
      src="/images/aeneid-demo-image-2.png"
    />
    <button
      className="show-in-viewer-button"
      disabled={false}
      href={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      onTouchTap={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "border": 10,
          "borderRadius": 2,
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": 36,
          "lineHeight": "36px",
          "margin": 0,
          "minWidth": 88,
          "muiPrepared": true,
          "outline": "none",
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "verticalAlign": null,
          "zIndex": 1,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <i
          className="mdi mdi-magnify"
          color="rgba(0, 0, 0, 0.87)"
          style={
            Object {
              "marginLeft": 12,
              "marginRight": 0,
              "verticalAlign": "middle",
            }
          }
        />
        <span
          style={
            Object {
              "fontSize": 14,
              "fontWeight": 500,
              "letterSpacing": 0,
              "muiPrepared": true,
              "paddingLeft": 8,
              "paddingRight": 16,
              "position": "relative",
              "textTransform": "uppercase",
              "verticalAlign": "middle",
            }
          }
        >
          Image Viewer
        </span>
      </div>
    </button>
  </div>
  <div
    className="media-description"
  >
    <span
      className="media-caption"
    >
      "Folio 45V". &lt;em&gt;The Vergilius Vaticanus&lt;/em&gt;. Vatican, Biblioteca Apostolica, Cod. Vat. lat. 3225). Rome, Italy. Ca 400 AD.
    </span>
  </div>
</div>
`;
